src/lib/POWER_DETECT/devPDET.cpp

```diff
@@ -29,41 +34,39 @@ static int start()
 
 static int timeout()
 {
-    if (!busyTransmitting) return PDET_BUSY_PERIOD;
+    if (!busyTransmitting) return PDET_BUSY_PERIODMS;
 
-    float newPdet = analogReadMilliVolts(GPIO_PIN_PA_PDET);
+    pdet_storage_t newPdetScaled = PDET_MV_SCALE(analogReadMilliVolts(GPIO_PIN_PA_PDET));
 
-    if (!busyTransmitting) return PDET_BUSY_PERIOD; // Check transmission did not stop during Pdet measurement.
+    if (!busyTransmitting) return PDET_BUSY_PERIODMS; // Check transmission did not stop during Pdet measurement.
 
-    if (!Pdet || currentPowerdBm != POWERMGNT::getPowerIndBm())
+    uint8_t targetPowerDbm = POWERMGNT::getPowerIndBm();
+    if (PdetMvScaled == 0 || lastTargetPowerdBm != targetPowerDbm)
     {
-        Pdet = newPdet;
-        currentPowerdBm = POWERMGNT::getPowerIndBm();
+        PdetMvScaled = newPdetScaled;
+        lastTargetPowerdBm = targetPowerDbm;
     }
     else
     {
-        Pdet = Pdet * 0.9 + newPdet * 0.1; // IIR filter
+        PdetMvScaled = (PdetMvScaled * 9 + newPdetScaled) / 10; // IIR filter
```

Comment:
Should this be
```cpp
PdetMvScaled = PDET_MV_DESCALE(PdetMvScaled * PDET_MV_SCALE(0.9) + newPdetScaled); // IIR filter
```

Reply:
Pretty sure Cap has that correct and everything is compared as scaled later.

Reply:
> Pretty sure Cap has that correct and everything is compared as scaled later.

He has it correct. But doing the scaled and descaled calls show the actual logic and then you don't have to think about it later on.

Reply:
Them both being in 10x scale is purely coincidental. The IIR just wants to take 90% of one value and 10% of the other value, which I've converted from 0.9 + 0.1 to (9 + 1)/10. The mV scale doesn't enter into that at all because the ratio is what the calculation is concerned with and therefore has no scale to it. The mV scale could be changed to 10000x and this calculation would still give the same values if it were 10000 scale or 10 scale. Increasing the scale of this calculation however can overflow so that's why it is only *9/10 instead of *0.9*scale.

Increasing the mV scale increases the precision of the remainder, but scaling this calculation does nothing but make the intermediate values larger.